### DISCUSSION TITLE
COUNT(DISTINCT(..)) support

### DISCUSSION THREAD

**Discussion by oeed** [diesel-rs/diesel#2713] [Ideas]
Currently (as far as I can find) there is no support for `COUNT(DISTINCT(column))` in Diesel. There's mention of it on [Gitter](https://gitter.im/diesel-rs/diesel?at=58bd8228de50490822519ced) over 4 years ago, I couldn't find the issue that was to be created, however. This [SO question](https://stackoverflow.com/questions/51852226/how-do-i-count-the-number-of-distinct-elements-in-an-array-column-with-diesel) from 2018 says to use a raw query.

Looking at the [implementation](https://docs.diesel.rs/src/diesel/expression/count.rs.html#7-29) of `count` and `count_star` it doesn't seem like it'd be particularly difficult so I'd be quite open to making a PR if welcome. It seems like `count_distinct` would be best (i.e. a single function rather than `count(distinct(column))` as distinct would never be used elsewhere.

Not overly familiar with whether the syntax is universally the same though, ANSI _seems_ to use `DISTINCT column` while MySQL uses `DISTINCT(column)`.

**Comment by weiznich** [comment]
> This SO question from 2018 says to use a raw query.

Seems like this suggests using a raw query because of the select as part of the from clause, not because of the `count(distinct(column))` clause.

Otherwise we are open to receive a PR here, as long as it includes tests + documentation and supports all build-in backends where this feature is provided by the database implementation.
